sourcesapi/sourcesapi.py

```diff
@@ -0,0 +1,309 @@
+"""Handles collecting data from source talkers.
+"""
+# Copyright 2012-2014 Anthony Beville
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import annotations
+
+import logging
+from typing import Callable
+
+from comicapi.genericmetadata import GenericMetadata
+from comictaggerlib.settings import ComicTaggerSettings
+from sourcesapi.resulttypes import ComicIssuesByVolumeResults, ComicSearchResults
+from sourcesapi.sourcebaseclass import SourceInfoError, SourceStaticOptions
+
+logger = logging.getLogger(__name__)
+
+
+# To signal image loaded etc.
+def list_fetch_complete(url_list: list[str]) -> None:
+    ...
+
+
+def url_fetch_complete(image_url: str, thumb_url: str | None) -> None:
+    ...
+
+
+class SourcesAPI:
```

Comment:
You defined `InfoSource` in sourcebaseclass.py use it
```suggestion
class SourcesAPI(InfoSource):
```

Reply:
I was looking at the wrong thing here....